server/src/main/java/org/elasticsearch/action/bulk/FailureStoreDocumentConverter.java

```diff
@@ -96,10 +99,28 @@ private static XContentBuilder createSource(
                 builder.field("type", ElasticsearchException.getExceptionName(unwrapped));
                 builder.field("message", unwrapped.getMessage());
                 builder.field("stack_trace", ExceptionsHelper.stackTrace(unwrapped));
-                // Further fields not yet tracked (Need to expose via specific exceptions)
-                // - pipeline
-                // - pipeline_trace
-                // - processor
+                // Try to find the IngestProcessorException somewhere in the stack trace. Since IngestProcessorException is package-private,
+                // we can't instantiate it in tests, so we'll have to check for the headers directly.
+                var ingestException = ExceptionsHelper.<ElasticsearchException>unwrapCausesAndSuppressed(
+                    exception,
+                    t -> t instanceof ElasticsearchException e && e.getHeaderKeys().stream().anyMatch(INGEST_EXCEPTION_HEADERS::contains)
```

Comment:
Nit: if both become sets we can use [`Sets.haveEmptyIntersection(e.getHeaderKeys(), INGEST_EXCEPTION_HEADERS) == false`](https://github.com/elastic/elasticsearch/blob/14ca58c9260e3d9bca86d561d254e84ed9266ed7/server/src/main/java/org/elasticsearch/common/util/set/Sets.java#L56).

Reply:
TIL, thanks :)